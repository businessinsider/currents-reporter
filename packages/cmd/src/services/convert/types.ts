export type ReportConfig = {
  framework: string;
  frameworkVersion: string | null;
  cliArgs?: Record<string, unknown>;
  frameworkConfig: Record<string, unknown>;
};

export interface Failure {
  message?: string;
  _?: string;
  type?: string;
}

export interface TestCase {
  name?: string;
  classname?: string;
  time?: string;
  failure?: (Failure | string)[];
  'system-out'?: string;
  'system-err'?: string;
  [key: string]: any;
}

export interface Property {
  name?: string;
  value?: string;
  _?: string;
}

export interface TestSuite {
  id?: string;
  name?: string;
  tests?: string;
  failures?: string;
  errors?: string;
  skipped?: string;
  assertions?: string;
  time?: string;
  timestamp?: string;
  file?: string;
  properties?: {
    property?: Property[];
  };
  'system-out'?: string;
  'system-err'?: string;
  testcase?: TestCase[];
}

export interface TestSuites {
  testsuites?: {
    name?: string;
    tests?: string;
    failures?: string;
    errors?: string;
    skipped?: string;
    assertions?: string;
    time?: string;
    timestamp?: string;
    testsuite?: TestSuite[];
    [key: string]: any;
  };
}
