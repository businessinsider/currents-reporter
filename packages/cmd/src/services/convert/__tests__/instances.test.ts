import { describe, it, expect, beforeEach } from 'vitest';
import { getInstanceMap } from '../postman';
import { InstanceReport } from 'types';

describe('getInstanceMap', () => {
  const xmlInput = `
    <testsuites name="Test Collection" tests="4" failures="0" errors="0" time="200">
      <testsuite name="Test Suite 1" timestamp="2024-12-20T22:12:47.937Z" tests="2" failures="0" errors="0" skipped="0" time="10">
        <testcase classname="path/to/file.test.ts" name="Test Case 1" time="3"/>
        <testcase classname="path/to/file.test.ts" name="Test Case 2" time="7"/>
      </testsuite>
      <testsuite name="Test Suite 2" timestamp="2024-12-20T22:12:57.937Z" tests="2" failures="0" errors="0" skipped="0" time="42">
        <testcase classname="path/to/file.test.ts" name="Test Case 3" time="12"/>
        <testcase classname="path/to/file.test.ts" name="Test Case 4" time="30"/>
      </testsuite>
    </testsuites>
  `;

  let instanceMap: Map<string, InstanceReport>;

  function getTestEntry(
    instanceMap: Map<string, InstanceReport>,
    testName: string
  ) {
    return Array.from(instanceMap.values()).find(instance =>
      instance.results.tests.some(test => test.title.includes(testName))
    );
  }

  beforeEach(async () => {
    instanceMap = await getInstanceMap(xmlInput);
  });

  it('returns a map of instances', () => {
    expect(instanceMap.size).toBe(2);
  });

  it.each([
    'Test Suite 1',
    'Test Suite 2',
  ])('matches snapshot for %s', testSuiteName => {
    expect(getTestEntry(instanceMap, testSuiteName)).toMatchSnapshot();
  });
});
