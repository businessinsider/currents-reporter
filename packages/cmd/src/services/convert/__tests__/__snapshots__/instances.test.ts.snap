// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getInstanceMap > matches snapshot for Test Suite 1 1`] = `
{
  "groupId": "Test Collection",
  "results": {
    "stats": {
      "failures": 0,
      "flaky": 0,
      "passes": 2,
      "pending": 0,
      "skipped": 0,
      "suites": 1,
      "tests": 2,
      "wallClockDuration": 10000,
      "wallClockEndedAt": "2024-12-20T22:12:57.937Z",
      "wallClockStartedAt": "2024-12-20T22:12:47.937Z",
    },
    "tests": [
      {
        "_t": 1734732767937,
        "attempts": [
          {
            "_s": "passed",
            "attempt": 0,
            "duration": 3000,
            "error": undefined,
            "errors": [],
            "parallelIndex": 1,
            "startTime": "2024-12-20T22:12:47.937Z",
            "status": "passed",
            "stderr": [],
            "stdout": [],
            "steps": [],
            "workerIndex": 1,
          },
        ],
        "expectedStatus": "passed",
        "isFlaky": false,
        "location": {
          "column": 1,
          "file": "",
          "line": 1,
        },
        "retries": 0,
        "state": "passed",
        "testId": "5ce2a25c0c0ac67a",
        "timeout": 0,
        "title": [
          "Test Suite 1",
          "Test Case 1",
        ],
      },
      {
        "_t": 1734732767937,
        "attempts": [
          {
            "_s": "passed",
            "attempt": 0,
            "duration": 7000,
            "error": undefined,
            "errors": [],
            "parallelIndex": 1,
            "startTime": "2024-12-20T22:12:47.937Z",
            "status": "passed",
            "stderr": [],
            "stdout": [],
            "steps": [],
            "workerIndex": 1,
          },
        ],
        "expectedStatus": "passed",
        "isFlaky": false,
        "location": {
          "column": 1,
          "file": "",
          "line": 1,
        },
        "retries": 0,
        "state": "passed",
        "testId": "865b7d9599598528",
        "timeout": 0,
        "title": [
          "Test Suite 1",
          "Test Case 2",
        ],
      },
    ],
  },
  "spec": "Test Suite 1",
  "startTime": "2024-12-20T22:12:47.937Z",
  "worker": {
    "parallelIndex": 1,
    "workerIndex": 1,
  },
}
`;

exports[`getInstanceMap > matches snapshot for Test Suite 2 1`] = `
{
  "groupId": "Test Collection",
  "results": {
    "stats": {
      "failures": 0,
      "flaky": 0,
      "passes": 2,
      "pending": 0,
      "skipped": 0,
      "suites": 1,
      "tests": 2,
      "wallClockDuration": 42000,
      "wallClockEndedAt": "2024-12-20T22:13:39.937Z",
      "wallClockStartedAt": "2024-12-20T22:12:57.937Z",
    },
    "tests": [
      {
        "_t": 1734732777937,
        "attempts": [
          {
            "_s": "passed",
            "attempt": 0,
            "duration": 12000,
            "error": undefined,
            "errors": [],
            "parallelIndex": 1,
            "startTime": "2024-12-20T22:12:57.937Z",
            "status": "passed",
            "stderr": [],
            "stdout": [],
            "steps": [],
            "workerIndex": 1,
          },
        ],
        "expectedStatus": "passed",
        "isFlaky": false,
        "location": {
          "column": 1,
          "file": "",
          "line": 1,
        },
        "retries": 0,
        "state": "passed",
        "testId": "fb352e5c3cc422e6",
        "timeout": 0,
        "title": [
          "Test Suite 2",
          "Test Case 3",
        ],
      },
      {
        "_t": 1734732777937,
        "attempts": [
          {
            "_s": "passed",
            "attempt": 0,
            "duration": 30000,
            "error": undefined,
            "errors": [],
            "parallelIndex": 1,
            "startTime": "2024-12-20T22:12:57.937Z",
            "status": "passed",
            "stderr": [],
            "stdout": [],
            "steps": [],
            "workerIndex": 1,
          },
        ],
        "expectedStatus": "passed",
        "isFlaky": false,
        "location": {
          "column": 1,
          "file": "",
          "line": 1,
        },
        "retries": 0,
        "state": "passed",
        "testId": "d7615409e164a73e",
        "timeout": 0,
        "title": [
          "Test Suite 2",
          "Test Case 4",
        ],
      },
    ],
  },
  "spec": "Test Suite 2",
  "startTime": "2024-12-20T22:12:57.937Z",
  "worker": {
    "parallelIndex": 1,
    "workerIndex": 1,
  },
}
`;
